airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -100,3 +102,20 @@ def patch_dag(session, dag_id, update_mask=None):
     setattr(dag, 'is_paused', patch_body['is_paused'])
     session.commit()
     return dag_schema.dump(dag)
+
+
+@security.requires_access([(permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG)])
+@provide_session
+def delete_dag(dag_id: str, session: Session):
+    """Delete the specific DAG."""
+    # TODO: This function is shared with the /delete endpoint used by the web
+    # UI, so we're reusing it to simplify maintenance. Refactor the function to
+    # another place when the experimental/legacy API is removed.
+    from airflow.api.common.experimental import delete_dag
+
+    try:
+        delete_dag.delete_dag(dag_id, session=session)
+    except DagNotFound:
+        raise NotFound(f"Dag with id: '{dag_id}' not found")
+
```

Comment:
Yes except that brings back the 409 issue 🙃

It also feels curious to me the web UI does not seem to handle this exception. Do we just get a 500 when that happens?

Reply:
It handles the exception see: https://github.com/apache/airflow/blob/6db73a60c9f6e0ba46712cf44b9f1e506450fcae/airflow/www/views.py#L1572-L1578

I think 400 will be good to return when that happens since the user can do something to resolve it?